codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -256,18 +270,24 @@ public final void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
      */
     protected void handlerRemoved0(ChannelHandlerContext ctx) throws Exception { }
 
+    /**
+     * Tells whether this handle has been removed from the {@link ChannelPipeline}.
+     * This method is useful because 1) a user can check if this handler is removed even if
+     * he or she does not have a reference to {@link ChannelHandlerContext} and 2) it is
+     * cheaper than {@link ChannelHandlerContext#isRemoved()}.
+     */
+    protected final boolean isRemoved() {
```

Comment:
@njhill honestly I think I would just remove this for now as it increase the API surface and the win on X86 should be quite small anyway.

Reply:
@normanmaurer yes on x86 the cost of the read is no different if/when it happens but the primary consideration is that the calls to this are interleaved with other non-volatile accesses which look ripe for reordering/elimination. We could have an extra boolean field instead if you are thinking to eliminate the state stuff at some point.

But sure I will remove it anyhow to keep the PR more focused, can always add/benchmark/discuss in another PR